### PR TITLE
fixed infinite loop with Makefile 3.81 and 4.3

### PR DESCRIPTION
**Pull request by jeonghanlee**
Please try, and let me know.

### CODE REVIEW

File: loopTestApp/Db/Makefile [hunk 1]
```diff
@@ -17,10 +17,11 @@ include $(TOP)/configure/RULES
 #----------------------------------------
 #  ADD RULES AFTER THIS LINE
 
-example.template: example.generated
+../O.Common/example.db: ../example.generated
```
Comment (ralphlange) [inline loopTestApp/Db/Makefile:20]: Well, I wouldn't put generated code in a source directory.
That template seems to be ARCH-independent, so should go into `O.Common`, shouldn't it?

Comment (jeonghanlee) [reply loopTestApp/Db/Makefile:20]: I thought you would mention this. I agree with you, @ralphlange. My trick is a quick and dirty solution.

**Comment by mp49** [comment]
So something like this?
```
../O.Common/example.db: ../O.Common/example.generated

../O.Common/example.generated:
        echo "# some more database logic" > $@
```

**Comment by mp49** [comment]
In practice the real module where this problem occured would only ever be compiled on Linux so this seems ok too:
```
../O.Common/example.db: example.generated

example.generated:
        echo "# some more database logic" > $@
```

**Comment by mp49** [comment]
Actually I couldn't get either of those working correctly. This seems to work though:
```
example.db.d: example.generated

example.generated:
        echo "# some more database logic" > $@
```
But, I am making an assumption about the name of the autogenerated `example.db.d`.

**Comment by simon-ess** [comment]
Another option that I think is a bit better is:
```make
include $(TOP)/configure/RULES
#----------------------------------------
#  ADD RULES AFTER THIS LINE

../example.template: $(COMMON_DIR)/example.generated

$(COMMON_DIR)/example.generated:
	echo "# some more database logic" > $(COMMON_DIR)/example.generated
```
You can also do
```make
../example.template: example.generated

example.generated:
	echo "# some more database logic" > example.generated
```
but I have to admit the example.generated seems more a "common" file, not an architecture-dependent one.

I will write a bit more about why I believe this is happening in the tech-talk thread.

**Pull request merged**